guides/migration/migrating-to-quarkus.adoc

```diff
@@ -0,0 +1,51 @@
+:guide-title: Migrating to Quarkus distribution
+:guide-summary: Migrate to the new Quarkus distribution from the legacy WildFly distribution
+
+In Keycloak 17 the default distribution is now powered by Quarkus, while the legacy WildFly powered distribution will still be around until June 2022 we highly recommend starting the migration as soon as possible.
+
+The new distribution introduces a number of breaking changes, including:
+
+* Configuring Keycloak has significantly changed
+* Quarkus is not an application server, but rather a framework to build applications
+* Custom providers are packaged and deployed differently
+
+Before undertaking the migration we highly recommend reading through the new https://www.keycloak.org/guides#server[Server Guides] to understand how to install and configure the new distribution.
+
+== Migrating configuration
+
+The WildFly distribution of Keycloak used complicated XML files for configuration, resulting in the need for a CLI tool (`jboss-cli`) to manipulate these files. These files also brought complication to upgrading, where an error-prone script was used to upgrade the config from a previous version.
+
+The new Quarkus powered distribution leverages a simple configuration file instead, with corresponding CLI arguments and environment variables as options, making it significantly easier to configure Keycloak. However, this results in the ability to automatically migrate the configuration from the previous distribution.
```

Comment:
```suggestion
The new Quarkus powered distribution leverages a simple configuration file instead, with corresponding CLI arguments and environment variables as options, making it significantly easier to configure Keycloak.

Since the WildFly distribution was too flexible in how much it could be configured and tuned, as well as the approach for configuration changing so much, it is not possible to automatically migrate the configuration from the old WildFly distribution to the new Quarkus distribution.
```